application/classes/Ushahidi/Formatter/Post/CSV.php

```diff
@@ -71,12 +76,45 @@ protected function generateCSVRecords($records)
 			foreach ($heading as $key => $value) {
 				$values[] = $this->getValueFromRecord($record, $key);
 			}
-			fputcsv($fp, $values);
+			fputcsv($stream, $values);
+		}
+
+		return $this->writeStreamToFS($stream);
+	}
+
+
+	private function writeStreamToFS($stream)
+	{
+
+		$filepath = implode('/', [
+			'csv',
+			$this->tmpfname,
+			]);
+
+		// Remove any leading slashes on the filename, path is always relative.
+		$filepath = ltrim($filepath, '/');
+
+		$extension = pathinfo($filepath, PATHINFO_EXTENSION);
+		
+		$mimeType = MimeType::detectByFileExtension($extension) ?: 'text/plain';
+		
+		$config = ['mimetype' => $mimeType];
+		
+		$this->fs->putStream($filepath, $stream, $config);
+
+		if (is_resource($stream)) {
+			fclose($stream);
```

Comment:
I think if you do this here it'll just be destroyed in each row because it uses tmpfile. Is that the idea? (sorry I'm still trying to understand everything here, this might be a dumb question)

Reply:
No it's not dumb, it's annoying how this works let me try to explain.

We write to a temp file managed by php, once we have completed writing to that file, we take it and write it to the CDN(rackspace, aws). The reason we have to do this is because you can not append to a file via the flyssytem lib on either rackspace or aws. Aws actually supports that but the lib doesn't.